### PR TITLE
fix: Remove user from managed event type when removing from org

### LINKED ISSUE #17363: [CAL-4627] Remove user from managed event type when removing from org
When a user is removed from an organization, we need to delete the assignment to the managed event type. I didn't test removing a member from a team, but this should also happen for teams.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4627](https://linear.app/calcom/issue/CAL-4627/remove-user-from-managed-event-type-when-removing-from-org)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

- Fixes #17363  
- Fixes CAL-4627

[Screencast from 2024-10-28 20-36-42.webm](https://github.com/user-attachments/assets/bdc1c8fe-41a8-423a-9c62-edf92afa4a05)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by anikdhabal — DISMISSED** [review]
LGTM!

File: packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts [hunk 1]
```diff
@@ -57,13 +57,29 @@ export async function bulkDeleteUsersHandler({ ctx, input }: BulkDeleteUsersHand
...
+        OR: [
+          {
+            parentId: currentUser.organizationId,
+          },
+          { id: currentUserOrgId },
+        ],
+      },
+    },
+  });
+
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts:75]: In the issue it is mentioned that you need to only remove the assignment but here you are deleting all event types

**Comment by keithwillcode** [comment]
@emrysal Could you please take a look at this one before we merge?

**Review by sean-brydon — DISMISSED** [review]
I think now would be a good time to ensure that deleting users does what we expect by having some test around it. This is a pretty high risk change so personally i'd like to see some introduced for this change.

**Comment by Amit91848** [comment]
> I think now would be a good time to ensure that deleting users does what we expect by having some test around it. This is a pretty high risk change so personally i'd like to see some introduced for this change.

Will add

**Review by Amit91848 — COMMENTED** [review]
Added test. Initially thought of breaking it down into multiple tests, but it would need a lot of duplicate code so added it in a single test, lmk if this should be changed.

**Comment by Amit91848** [comment]
> I think now would be a good time to ensure that deleting users does what we expect by having some test around it. This is a pretty high risk change so personally i'd like to see some introduced for this change.

Will add tests for team member removal in a follow up PR, or else this would get too big

File: packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts [hunk 2]
```diff
@@ -57,13 +57,55 @@ export async function bulkDeleteUsersHandler({ ctx, input }: BulkDeleteUsersHand
...
+      userId: {
+        in: input.userIds,
+      },
+      parent: {
+        team: {
+          OR: [
+            {
+              parentId: currentUserOrgId,
+            },
+            { id: currentUserOrgId },
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts:71]: Do we really need this? as  

```
owner       User?   @relation("owner", fields: [userId], references: [id], onDelete: Cascade)
  userId      Int?
```
  
 onDelete: Cascade is already there

**Review by Udit-takkar — COMMENTED** [review]
Tested everything and it works fine. ~left one comment https://github.com/calcom/cal.com/pull/17369/files#r1863830580~

File: packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts (hunk 2 above)
Comment (Amit91848) [reply packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts:71]: The `onDelete: Cascade` rule would trigger deletion when the user entry is being removed from the database. However, in this case, we are **not** deleting the user from the database; we are only removing the relationship between the user and the organization

https://github.com/calcom/cal.com/blob/6e22e06fa90f578693c0c0e20ccae518fec09338/packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts#L45-L58

The user record still exists in the database. Cascade only works when the user row itself is deleted @Udit-takkar

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**